### ISSUE TITLE
Using new scipy stats sampling methods

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#23785]
There are new methods in scipy for doing sampling of statistical distributions. See the guide at https://docs.scipy.org/doc/scipy/tutorial/stats/sampling.html. There are different methods depending on whether you want a fast setup speed or fast sampling speed. 

We should use this instead of `rvs`. I spoke with @mdhaber and @tupui here at the SciPy conference sprints, and they said we should be using this new API. They are no longer interested in supporting the old API. It does require SciPy 1.8 or newer, but I think that's fine, since it's an optional dependency anyway.

**Comment by oscarbenjamin** [comment]
Also the sampling can be much faster if it is based on cdf or inverse cdf rather than pdf. SymPy should maybe try to compute those.

**Comment by tupui** [comment]
Yes these methods are fast and in the tutorial we have a nice table explaining which method is faster depending on whether you need sampling speed or init speed.

**Comment by samithkavishke** [comment]
Is this still open?
